warehouse/ingest-core/src/main/java/datawave/util/flag/FlagMaker.java

```diff
@@ -502,13 +499,17 @@ protected File write(Collection<InputFile> flagging, FlagDataTypeConfig fc, Stri
             throw new IOException("Unable to create flag file " + f);
         }
         
+        final FileSystem fs = getHadoopFS();
+        final FlagMetrics metrics = new FlagMetrics(fs, fc.isCollectMetrics());
```

Comment:
So now we have created a new metrics object, and updated counters therein, but never write the metrics out.  So 1) are these metrics even needed and 2) if they are then we should be writing the metrics out.

Reply:
The metrics get written out on line 475 in FlagMaker.java .  I added the FlagMetrics object as a parameter in the write and writeFlagFile functions so it only has to get initialized once in the processFlags function.  I checked the call stacks for those functions and it looks like we are all good now.